### ISSUE TITLE
debugbot can't connect to wonnode (live)

### ISSUE THREAD

**Issue by fkleedorfer** [researchstudio-sat/webofneeds#1160]
Seeing this in the won node's logs:
```
17.7.17 09:34:35.127 [ActiveMQ BrokerService[wonBroker] Task-2173] ERROR o.a.a.broker.TransportConnector - Could not accept connection from tcp://172.18.0.1:43918: javax.net.ssl.SSLHandshakeException: Received fatal alert: certificate_unknown
```
and this in the debugbot's:
```
17.7.17 09:36:36.490 [Thread-54] WARN  w.o.m.OwnerWonMessageSenderJMSBased - Could not register with default won node https://node.matchat.org/won/resource. Try again later ... (reason is logged at loglevel 'DEBUG')
17.7.17 09:36:36.589 [Camel (wonOwnerCamel) thread #13 - TemporaryQueueReplyManager[OwnerProtocol.in]] ERROR o.a.c.c.j.DefaultJmsMessageListenerContainer - Could not refresh JMS Connection for destination 'temporary' - retrying using FixedBackOff{interval=5000, currentAttempts=83, maxAttempts=unlimited}. Cause: Could not connect to broker URL: ssl://node.matchat.org:61616. Reason: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
```

Somehow the WoN node forgot the debugbot's certificate, or the debugbot created a new one

**Comment by fkleedorfer** [comment]
not fixed by the re-registering code

**Comment by fkleedorfer** [comment]
Had a look at the content of the wonnode trust store.  Just a quick reminder, if this should come up again: 

Doing this, in principle, is documented  in [Cryptographic Keys and Certificates](../tree/master/documentation/installation-cryptographic-keys-and-certificates.md), but to do it inside the wonnode container requires more work:

1. copy the bouncycastle libs from the tomcat libs to the jre/lib/ext folder:
`cp /usr/local/tomcat/lib/bcp* /usr/lib/jvm/java-8-openjdk-amd64/jre/lib/ext/`
this allows keytool to read the bouncycastle keystore format
2. use keytool to list the keystore contents:
`keytool -list -v -providerclass org.bouncycastle.jce.provider.BouncyCastleProvider -storetype UBER -storepass <YOUR_KEYSTORE_PASSWORD> -keystore t-keystore.jks`

you're welcome.

**Comment by fkleedorfer** [comment]
Inspecting the node's trust store shows that the certificates of owner and debugbot are stored there. 

The question is why activemq does not pick them up during the TLS handshake

**Comment by fkleedorfer** [comment]
Here is the full exception, this one actually from the matcher trying to connect to the WoN node:
```
17.7.17 15:40:18.636 [ActiveMQ BrokerService[wonBroker] Task-9] DEBUG o.a.a.broker.TransportConnector - Reason: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.ssl.Alerts.getSSLException(Alerts.java:192) ~[na:1.8.0_121]
        at sun.security.ssl.SSLSocketImpl.fatal(SSLSocketImpl.java:1949) ~[na:1.8.0_121]
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:302) ~[na:1.8.0_121]
        at sun.security.ssl.Handshaker.fatalSE(Handshaker.java:296) ~[na:1.8.0_121]
        at sun.security.ssl.ServerHandshaker.clientCertificate(ServerHandshaker.java:1906) ~[na:1.8.0_121]
        at sun.security.ssl.ServerHandshaker.processMessage(ServerHandshaker.java:233) ~[na:1.8.0_121]
        at sun.security.ssl.Handshaker.processLoop(Handshaker.java:1026) ~[na:1.8.0_121]
        at sun.security.ssl.Handshaker.process_record(Handshaker.java:961) ~[na:1.8.0_121]
        at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:1062) ~[na:1.8.0_121]
        at sun.security.ssl.SSLSocketImpl.performInitialHandshake(SSLSocketImpl.java:1375) ~[na:1.8.0_121]
        at sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:928) ~[na:1.8.0_121]
        at sun.security.ssl.AppInputStream.read(AppInputStream.java:105) ~[na:1.8.0_121]
        at org.apache.activemq.transport.tcp.TcpBufferedInputStream.fill(TcpBufferedInputStream.java:50) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.tcp.TcpTransport$2.fill(TcpTransport.java:604) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.tcp.TcpBufferedInputStream.read(TcpBufferedInputStream.java:58) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.tcp.TcpTransport$2.read(TcpTransport.java:589) ~[activemq-client-5.8.0.jar:5.8.0]
        at java.io.DataInputStream.readInt(DataInputStream.java:387) ~[na:1.8.0_121]
        at org.apache.activemq.openwire.OpenWireFormat.unmarshal(OpenWireFormat.java:275) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.tcp.TcpTransport.readCommand(TcpTransport.java:221) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.tcp.TcpTransport.doRun(TcpTransport.java:213) ~[activemq-client-5.8.0.jar:5.8.0]
        at org.apache.activemq.transport.tcp.TcpTransport.run(TcpTransport.java:196) ~[activemq-client-5.8.0.jar:5.8.0]
        at java.lang.Thread.run(Thread.java:745) [na:1.8.0_121]
Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:387) ~[na:1.8.0_121]
        at sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:292) ~[na:1.8.0_121]
        at sun.security.validator.Validator.validate(Validator.java:260) ~[na:1.8.0_121]
        at sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:324) ~[na:1.8.0_121]
        at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:227) ~[na:1.8.0_121]
        at sun.security.ssl.X509TrustManagerImpl.checkClientTrusted(X509TrustManagerImpl.java:99) ~[na:1.8.0_121]
        at won.cryptography.ssl.TrustManagerWrapperWithTrustService.checkClientTrusted(TrustManagerWrapperWithTrustService.java:41) ~[won-core-0.2-SNAPSHOT.jar:na]
        at sun.security.ssl.AbstractTrustManagerWrapper.checkClientTrusted(SSLContextImpl.java:977) ~[na:1.8.0_121]
        at sun.security.ssl.ServerHandshaker.clientCertificate(ServerHandshaker.java:1888) ~[na:1.8.0_121]
        ... 17 common frames omitted
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141) ~[na:1.8.0_121]
        at sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126) ~[na:1.8.0_121]
        at java.security.cert.CertPathBuilder.build(CertPathBuilder.java:280) ~[na:1.8.0_121]
        at sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:382) ~[na:1.8.0_121]
        ... 25 common frames omitted
```

**Comment by fkleedorfer** [comment]
Fixed: Duplicate of #891

We were blindsided by the letsencrypt certificate being updating automatically, Possilbly through a cronjob we did not see. 
Also, we interpreted the error message the wrong way. The server actually complained about the client not accepting the server certificate.
